astropy/io/fits/hdu/base.py

```diff
@@ -457,7 +457,7 @@ def block_iter(nbytes):
         hdu._data_size = size + _pad_length(size)
 
         # Checksums are not checked on invalid HDU types
-        if checksum and checksum != 'remove' and isinstance(hdu, _ValidHDU):
+        if isinstance(hdu, _ValidHDU):
```

Comment:
I'm confused, why does it need to be set if one doesn't want to compare it to the computed checksum? It's a private attribute so anything except the class or subclasses shouldn't really be using it. I mean the script is basically just asking **if** there's a `CHECKSUM` in the header, right?